modules/lang-painless/src/main/java/org/elasticsearch/painless/ir/AssignmentNode.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.painless.ir;
+
+
+import org.elasticsearch.painless.ClassWriter;
+import org.elasticsearch.painless.DefBootstrap;
+import org.elasticsearch.painless.Globals;
+import org.elasticsearch.painless.MethodWriter;
+import org.elasticsearch.painless.Operation;
+import org.elasticsearch.painless.lookup.PainlessCast;
+import org.elasticsearch.painless.lookup.PainlessLookupUtility;
+import org.elasticsearch.painless.lookup.def;
+
+public class AssignmentNode extends BinaryNode {
+
+    /* ---- begin node data ---- */
+
+    private boolean pre;
+    private boolean post;
+    private Operation operation;
+    private boolean read;
+    private boolean cat;
```

Comment:
What is a `cat` here?

Reply:
`cat` is short for concatenation for `String` types. These names are copied directly from their equivalent "user" node, so I would prefer to leave this for now as mechanical, but renaming should be a future change.

Reply:
It's clear when reading the code further down, but a word or two as a comment would be nice.

Reply:
Added a comment here and in BinaryMathNode that also does concatenations.